lib/cocoapods/installer/project_cache/project_cache_analyzer.rb

```diff
@@ -87,7 +87,7 @@ def analyze
           dirty_targets = compute_dirty_targets(pod_targets + aggregate_targets)
           dirty_pod_targets, dirty_aggregate_targets = dirty_targets.partition { |target| target.is_a?(PodTarget) }
 
-          pod_targets_to_generate = changed_pod_targets + added_pod_targets + dirty_pod_targets
+          pod_targets_to_generate = (changed_pod_targets + added_pod_targets + dirty_pod_targets).uniq
```

Comment:
does `aggregate_target_to_generate` also need to be de-duped?

Reply:
Nope. `aggregate_targets` are already du-duped by the resolves and returns all if any are dirty.